### DISCUSSION TITLE
Publishing to hex.pm

### DISCUSSION THREAD

**Discussion by lnr0626** [visciang/telegram#179] [General]
Howdy again,

I was wondering why this library isn't published to hex.pm. The main benefits over the current approach would be that the documentation would be automatically published to hexdocs.pm; though I don't think there's many benefits beyond that. This is almost purely curiosity since your current approach of using git tags definitely works.

**Comment by visciang** [comment]
I started this project a few years ago as a way to control and interact with a Raspberry Pi at home, using cameras and sensors. At the time, I didn’t publish it to Hex.pm because I didn’t want to add more "weekend projects" to the package manager.

But as the project grew, it turned into something more than just a hobby. I thought about publishing it, but found out the name "telegram" was already taken (https://hex.pm/packages/telegram) by a project that only lasted 12 commits over a month and was published after just three commits.

Publishing on Hex.pm has some perks though, beyond just getting the docs out there. It would integrate the package into the dependency tree, allowing for semantic versioning and smoother dependency resolution, especially if it ends up as a sub-dependency in other projects (not really relevant here, I think).
